app/services/katello/pulp3/repository.rb

```diff
@@ -111,6 +115,17 @@ def get_distribution(href)
         nil
       end
 
+      def update_distributions
+        paths.values.select { |prefix, _path| prefix == 'http' }.each do |path|
+          dists = lookup_distributions(base_path: path.sub(/^\//, ''))
+          delete_distribution(dists.first._href) if dists.first
+
+          if repo.root.unprotected
+            create_distribution(prefix, path)
+          end
```

Comment:
also this if statement is inside the loop, which means it will try to create the distribution  (http or https) if repo.root.unprotected is set, which isn't quite right.  We'd want to create a http distribution if unprotected is set, we always want an http distribution (which should already have been created when this has been called).

Reply:
I've re-read this a couple of times, but I'm not parsing it correctly. I'll ask about it on today's call.

Reply:
as we discussed yesterday, this is a bad idea and doesn't work with on_demand syncs (when we implement those).  So we'll want to update the remote as well on repository update

Reply:
Understand re the remote updates, but still confused about the comment re: distributions and what needs to be updated.

Reply:
whoops i commented in the wrong spot :)

All repositories are exposed via https, but some are also exposed via http via the 'unprotected' attribute on the Katello::Repository  object.  In pulp3 we are currently controller this by creating 1 or 2 distributions depending on if this attribute is set.   Each distribution is prefixed by either /http or /https.  We will always create one prefixed by /https, and will create one prefixed by /http IF the unprotected attribute is set on the repository.  

So for update, we need to either delete or create the distribution depending on if the unprotected flag is set or not, and if the /http distribution has been created or not.

Reply:
One point of confusion I just realized was causing me problems: the DistributionReference "path" attribute includes the prefix, whereas the pulp3 model Distribution doesn't include the prefix as part of the (relative) "path" attribute.

Reply:
""'whereas the pulp3 model Distribution doesn't include the prefix as part of the (relative) "path" attribute.""

I'm not sure I quite follow, it should be getting formed here, and then set on the distribution at creation:  https://github.com/Katello/katello/blob/master/app/services/katello/pulp3/repository.rb#L91-L97